tests/core/Resolver.spec.ts

```diff
@@ -377,6 +377,31 @@ describe('Resolver', () => {
 
       done();
     });
+
+    it('should stop applying if did state becomes undfined, and return the previous state.', async (done) => {
+      // Setting up initial DID state for the test.
+      const createOperationData = await OperationGenerator.generateAnchoredCreateOperation({ transactionTime: 1, transactionNumber: 1, operationIndex: 1 });
+      const initialDidState = await operationProcessor.apply(createOperationData.anchoredOperationModel, undefined);
+
+      const recoveryOperation1Data = await OperationGenerator.generateRecoverOperation({
+        didUniqueSuffix: createOperationData.createOperation.didUniqueSuffix,
+        recoveryPrivateKey: createOperationData.recoveryPrivateKey
+      });
+
+      const recoveryOperation1 = OperationGenerator.createAnchoredOperationModelFromOperationModel(recoveryOperation1Data.recoverOperation, 2, 2, 2);
+
+      const recoveryCommitValueToOperationMap = new Map<string, AnchoredOperationModel[]>();
+      const nextRecoveryCommitment = createOperationData.createOperation.suffixData.recoveryCommitment;
+      recoveryCommitValueToOperationMap.set(nextRecoveryCommitment, [recoveryOperation1]);
+
+      spyOn(resolver as any, 'applyFirstValidOperation').and.returnValue(Promise.resolve(undefined));
```

Comment:
I may have complicated thing unnecessarily by allowing return of undefined.... I wonder if it is possible to just force return `DidState` as we know a previous state must exist...